pandas/core/indexes/period.py

```diff
@@ -248,21 +240,7 @@ def _engine(self):
         return self._engine_type(lambda: self, len(self))
 
     @classmethod
-    def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        """
-        Values can be any type that can be coerced to Periods.
-        Ordinals in an ndarray are fastpath-ed to `_from_ordinals`
-        """
-        if not is_integer_dtype(values):
-            values = np.array(values, copy=False)
-            if len(values) > 0 and is_float_dtype(values):
-                raise TypeError("PeriodIndex can't take floats")
-            return cls(values, name=name, freq=freq, **kwargs)
-
-        return cls._from_ordinals(values, name, freq, **kwargs)
```

Comment:
If keeping this now, I think it will be clearer what the changes are once we actually split index/array

Reply:
I don't think it matters too much either way. @jbrockmendel does reverting this change cause issues with master? Or does the "old" `PeriodIndex._simple_new` do the right thing, at the cost of code duplication? My PeriodArray PR is going to change this again anyway

Reply:
It has changed since joris's comment above, but this is just a matter of code duplication.

> My PeriodArray PR is going to change this again anyway

I'm kind of hoping I can simplify these a good deal further before you finish up with SparseEA